lizmap/modules/admin_api/controllers/repository_rights_rest.classic.php

```diff
@@ -9,76 +9,144 @@
+            throw new Exception("The repository doesn't exist !", 404);
+        }
+
+        return RepositoryRightsService::getRights($repo->getKey());
+    }
+
+    /**
+     * Adds a specific right for a group to a repository.
+     *
+     * @return object the response object containing the details of the operation,
+     *                or an error message if the operation fails
      */
     public function post(): object
     {
+        /** @var jResponseJson $rep */
         $rep = $this->getResponse('json');
 
         if (!Credentials::handle()) {
             return Error::setError($rep, 401);
         }
 
-        $repo = $this->param('repo');
+        $key = $this->param('repo');
+        $group = $this->param('group');
+        $right = $this->param('right');
 
-        if (!lizmap::getRepository($repo)) {
-            return Error::setError($rep, 404, "The repository '{$repo}' doesn't exists.");
+        try {
+            Utils::verifyVars($group, $right, $key);
+        } catch (Exception $e) {
+            return Error::setError($rep, $e->getCode(), $e->getMessage());
         }
 
-        $action = $this->param('method');
+        $isAdded = false;
 
-        if ($action == 'add' || $action == 'remove') {
-            $rep = $this->editRights($rep);
-        } else {
-            return Error::setError($rep, 501, "'{$action}' is not implemented.");
+        try {
+            $isAdded = jAcl2DbManager::addRight($group, $right, $key);
+        } catch (Exception $e) {
+            return Error::setError($rep, 500, $e->getMessage());
         }
 
+        $rep->data = array(
+            'key' => $key,
+            'group' => $group,
+            'right' => $right,
+            'isAdded' => $isAdded,
+        );
+
         return $rep;
     }
 
     /**
-     * Modifies access rights for a specified group on a repository.
-     *
-     * @param object $rep the JSON response object to be modified
+     * Removes a specific right for a group to a repository.
      *
-     * @return object the modified JSON response object
+     * @return object the response object containing the details of the operation,
+     *                or an error message if the operation fails
      */
-    private function editRights(object $rep): object
+    public function delete(): object
     {
+        /** @var jResponseJson $rep */
+        $rep = $this->getResponse('json');
+
+        if (!Credentials::handle()) {
+            return Error::setError($rep, 401);
+        }
+
         $key = $this->param('repo');
         $group = $this->param('group');
         $right = $this->param('right');
 
-        $isValid = Utils::verifyVars($group, $right, $key);
-
-        if (!$isValid['bool']) {
-            return Error::setError($rep, 400, $isValid['message']);
+        try {
+            Utils::verifyVars($group, $right, $key);
+        } catch (Exception $e) {
+            return Error::setError($rep, $e->getCode(), $e->getMessage());
```

Comment:
```suggestion
        try {
            Utils::verifyVars($group, $right, $key);
        } catch (ApiException $e) {
            return Error::setError($rep, $e->getCode(), $e->getMessage());
        } catch (Exception $e) {
            return Error::setError($rep, 500, $e->getMessage());
```

Once `LizmapApi\ApiException` is defined